[2025-07-26T13:18:40.645+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-26T13:18:40.875+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: sample_ETL.load_order_and_delivery_data manual__2025-07-26T13:18:35.449704+00:00 [queued]>
[2025-07-26T13:18:40.888+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: sample_ETL.load_order_and_delivery_data manual__2025-07-26T13:18:35.449704+00:00 [queued]>
[2025-07-26T13:18:40.888+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-07-26T13:18:40.917+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): load_order_and_delivery_data> on 2025-07-26 13:18:35.449704+00:00
[2025-07-26T13:18:40.927+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=98) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-26T13:18:40.929+0000] {standard_task_runner.py:63} INFO - Started process 102 to run task
[2025-07-26T13:18:40.928+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'sample_ETL', 'load_order_and_delivery_data', 'manual__2025-07-26T13:18:35.449704+00:00', '--job-id', '88', '--raw', '--subdir', 'DAGS_FOLDER/sample_ETL.py', '--cfg-path', '/tmp/tmpt9ccsh9m']
[2025-07-26T13:18:40.931+0000] {standard_task_runner.py:91} INFO - Job 88: Subtask load_order_and_delivery_data
[2025-07-26T13:18:41.002+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/settings.py:195 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2025-07-26T13:18:41.110+0000] {task_command.py:426} INFO - Running <TaskInstance: sample_ETL.load_order_and_delivery_data manual__2025-07-26T13:18:35.449704+00:00 [running]> on host 88b2e53952eb
[2025-07-26T13:18:41.463+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='sample_ETL' AIRFLOW_CTX_TASK_ID='load_order_and_delivery_data' AIRFLOW_CTX_EXECUTION_DATE='2025-07-26T13:18:35.449704+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-07-26T13:18:35.449704+00:00'
[2025-07-26T13:18:41.465+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-26T13:18:41.481+0000] {logging_mixin.py:188} INFO - ðŸš€ Starting ETL with JSON files
[2025-07-26T13:18:41.486+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-producer-1, node_id=bootstrap-0 host=kafka:9092 <connecting> [IPv4 ('172.18.0.7', 9092)]>: connecting to kafka:9092 [('172.18.0.7', 9092) IPv4]
[2025-07-26T13:18:41.536+0000] {conn.py:618} INFO - <BrokerConnection client_id=kafka-python-producer-1, node_id=bootstrap-0 host=kafka:9092 <checking_api_versions_recv> [IPv4 ('172.18.0.7', 9092)]>: Broker version identified as 2.6
[2025-07-26T13:18:41.538+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-producer-1, node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Connection complete.
[2025-07-26T13:18:41.652+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-producer-1, node_id=1 host=kafka:9092 <connecting> [IPv4 ('172.18.0.7', 9092)]>: connecting to kafka:9092 [('172.18.0.7', 9092) IPv4]
[2025-07-26T13:18:41.653+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-producer-1, node_id=1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Connection complete.
[2025-07-26T13:18:41.654+0000] {conn.py:945} INFO - <BrokerConnection client_id=kafka-python-producer-1, node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Closing connection. 
[2025-07-26T13:18:41.707+0000] {logging_mixin.py:188} INFO - âœ… Produced 124 records to topic 'orders_topic'
[2025-07-26T13:18:41.708+0000] {logging_mixin.py:188} INFO - âœ… Orders producer completed
[2025-07-26T13:18:41.710+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <connecting> [IPv4 ('172.18.0.7', 9092)]>: connecting to kafka:9092 [('172.18.0.7', 9092) IPv4]
[2025-07-26T13:18:41.715+0000] {conn.py:618} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <checking_api_versions_recv> [IPv4 ('172.18.0.7', 9092)]>: Broker version identified as 2.6
[2025-07-26T13:18:41.716+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Connection complete.
[2025-07-26T13:18:41.717+0000] {subscription_state.py:181} INFO - Updating subscribed topics to: ('orders_topic',)
[2025-07-26T13:18:41.763+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connecting> [IPv4 ('172.18.0.7', 9092)]>: connecting to kafka:9092 [('172.18.0.7', 9092) IPv4]
[2025-07-26T13:18:41.764+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Connection complete.
[2025-07-26T13:18:41.765+0000] {conn.py:945} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Closing connection. 
[2025-07-26T13:18:41.871+0000] {cluster.py:401} INFO - Coordinator for group/etl_group_orders is ('coordinator-1', 'kafka', 9092, None)
[2025-07-26T13:18:41.872+0000] {base.py:788} INFO - Discovered coordinator coordinator-1 for group etl_group_orders
[2025-07-26T13:18:41.873+0000] {base.py:848} INFO - Starting new heartbeat thread
[2025-07-26T13:18:41.875+0000] {consumer.py:365} INFO - Revoking previously assigned partitions () for group etl_group_orders
[2025-07-26T13:18:41.876+0000] {base.py:419} INFO - Failed to join group etl_group_orders: NodeNotReadyError: coordinator-1
[2025-07-26T13:18:41.877+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=kafka:9092 <connecting> [IPv4 ('172.18.0.7', 9092)]>: connecting to kafka:9092 [('172.18.0.7', 9092) IPv4]
[2025-07-26T13:18:41.879+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Connection complete.
[2025-07-26T13:18:41.980+0000] {base.py:532} INFO - (Re-)joining group etl_group_orders
[2025-07-26T13:18:42.025+0000] {base.py:637} INFO - Received member id kafka-python-2.2.15-3ebb4d4a-667e-4534-941b-e7b7003e8aa0 for group etl_group_orders; will retry join-group
[2025-07-26T13:18:42.026+0000] {base.py:419} INFO - Failed to join group etl_group_orders: [Error 79] MemberIdRequiredError
[2025-07-26T13:18:42.027+0000] {base.py:532} INFO - (Re-)joining group etl_group_orders
[2025-07-26T13:18:45.068+0000] {base.py:594} INFO - Successfully joined group etl_group_orders <Generation 3 (member_id: kafka-python-2.2.15-3ebb4d4a-667e-4534-941b-e7b7003e8aa0, protocol: range)>
[2025-07-26T13:18:45.070+0000] {base.py:596} INFO - Elected group leader -- performing partition assignments using range
[2025-07-26T13:18:45.093+0000] {subscription_state.py:254} INFO - Updated partition assignment: [('orders_topic', 0)]
[2025-07-26T13:18:45.094+0000] {consumer.py:249} INFO - Setting newly assigned partitions (('orders_topic', 0),) for group etl_group_orders
[2025-07-26T13:18:57.198+0000] {logging_mixin.py:188} INFO - âœ… Inserted 124 total records into orders
[2025-07-26T13:18:57.226+0000] {base.py:1052} INFO - Stopping heartbeat thread
[2025-07-26T13:18:57.230+0000] {base.py:891} INFO - Leaving consumer group (etl_group_orders).
[2025-07-26T13:18:57.343+0000] {base.py:906} INFO - LeaveGroup request for group etl_group_orders returned successfully
[2025-07-26T13:18:57.345+0000] {conn.py:945} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Closing connection. 
[2025-07-26T13:18:57.348+0000] {fetcher.py:786} INFO - Fetch to node 1 failed: Cancelled: <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>
[2025-07-26T13:18:57.350+0000] {conn.py:945} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Closing connection. 
[2025-07-26T13:18:57.353+0000] {logging_mixin.py:188} INFO - âœ… Orders consumer completed
[2025-07-26T13:19:02.518+0000] {logging_mixin.py:188} INFO - âœ… Produced 124 records to topic 'delivery_topic'
[2025-07-26T13:19:02.521+0000] {logging_mixin.py:188} INFO - âœ… Delivery producer completed
[2025-07-26T13:19:02.529+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <connecting> [IPv4 ('172.18.0.7', 9092)]>: connecting to kafka:9092 [('172.18.0.7', 9092) IPv4]
[2025-07-26T13:19:02.552+0000] {conn.py:618} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <checking_api_versions_recv> [IPv4 ('172.18.0.7', 9092)]>: Broker version identified as 2.6
[2025-07-26T13:19:02.555+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Connection complete.
[2025-07-26T13:19:02.560+0000] {subscription_state.py:181} INFO - Updating subscribed topics to: ('delivery_topic',)
[2025-07-26T13:19:02.642+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connecting> [IPv4 ('172.18.0.7', 9092)]>: connecting to kafka:9092 [('172.18.0.7', 9092) IPv4]
[2025-07-26T13:19:02.643+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Connection complete.
[2025-07-26T13:19:02.644+0000] {conn.py:945} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Closing connection. 
[2025-07-26T13:19:02.755+0000] {cluster.py:401} INFO - Coordinator for group/etl_group_delivery is ('coordinator-1', 'kafka', 9092, None)
[2025-07-26T13:19:02.757+0000] {base.py:788} INFO - Discovered coordinator coordinator-1 for group etl_group_delivery
[2025-07-26T13:19:02.760+0000] {base.py:848} INFO - Starting new heartbeat thread
[2025-07-26T13:19:02.765+0000] {consumer.py:365} INFO - Revoking previously assigned partitions () for group etl_group_delivery
[2025-07-26T13:19:02.769+0000] {base.py:419} INFO - Failed to join group etl_group_delivery: NodeNotReadyError: coordinator-1
[2025-07-26T13:19:02.775+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=kafka:9092 <connecting> [IPv4 ('172.18.0.7', 9092)]>: connecting to kafka:9092 [('172.18.0.7', 9092) IPv4]
[2025-07-26T13:19:02.780+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Connection complete.
[2025-07-26T13:19:02.884+0000] {base.py:532} INFO - (Re-)joining group etl_group_delivery
[2025-07-26T13:19:02.894+0000] {base.py:637} INFO - Received member id kafka-python-2.2.15-4d002e1a-8116-4fcc-8ecf-5f1d28ec1267 for group etl_group_delivery; will retry join-group
[2025-07-26T13:19:02.897+0000] {base.py:419} INFO - Failed to join group etl_group_delivery: [Error 79] MemberIdRequiredError
[2025-07-26T13:19:02.900+0000] {base.py:532} INFO - (Re-)joining group etl_group_delivery
[2025-07-26T13:19:05.924+0000] {base.py:594} INFO - Successfully joined group etl_group_delivery <Generation 3 (member_id: kafka-python-2.2.15-4d002e1a-8116-4fcc-8ecf-5f1d28ec1267, protocol: range)>
[2025-07-26T13:19:05.927+0000] {base.py:596} INFO - Elected group leader -- performing partition assignments using range
[2025-07-26T13:19:05.946+0000] {subscription_state.py:254} INFO - Updated partition assignment: [('delivery_topic', 0)]
[2025-07-26T13:19:05.947+0000] {consumer.py:249} INFO - Setting newly assigned partitions (('delivery_topic', 0),) for group etl_group_delivery
[2025-07-26T13:19:18.069+0000] {logging_mixin.py:188} INFO - âœ… Inserted 124 total records into delivery
[2025-07-26T13:19:18.089+0000] {base.py:1052} INFO - Stopping heartbeat thread
[2025-07-26T13:19:18.092+0000] {base.py:891} INFO - Leaving consumer group (etl_group_delivery).
[2025-07-26T13:19:18.102+0000] {base.py:906} INFO - LeaveGroup request for group etl_group_delivery returned successfully
[2025-07-26T13:19:18.103+0000] {conn.py:945} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Closing connection. 
[2025-07-26T13:19:18.105+0000] {fetcher.py:786} INFO - Fetch to node 1 failed: Cancelled: <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>
[2025-07-26T13:19:18.106+0000] {conn.py:945} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=kafka:9092 <connected> [IPv4 ('172.18.0.7', 9092)]>: Closing connection. 
[2025-07-26T13:19:18.107+0000] {logging_mixin.py:188} INFO - âœ… Delivery consumer completed
[2025-07-26T13:19:18.108+0000] {logging_mixin.py:188} INFO - âœ… Task2 of ETL completed
[2025-07-26T13:19:18.109+0000] {python.py:237} INFO - Done. Returned value was: None
[2025-07-26T13:19:18.111+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-26T13:19:18.164+0000] {taskinstance.py:1206} INFO - Marking task as SUCCESS. dag_id=sample_ETL, task_id=load_order_and_delivery_data, run_id=manual__2025-07-26T13:18:35.449704+00:00, execution_date=20250726T131835, start_date=20250726T131840, end_date=20250726T131918
[2025-07-26T13:19:18.235+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2025-07-26T13:19:18.312+0000] {taskinstance.py:3498} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2025-07-26T13:19:18.329+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
